x-pack/test/cases_api_integration/security_and_spaces/tests/common/comments/delete_comments.ts

```diff
@@ -0,0 +1,480 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { CommentType } from '@kbn/cases-plugin/common';
+import { ALERT_CASE_IDS } from '@kbn/rule-data-utils';
+import {
+  createSecuritySolutionAlerts,
+  getAlertById,
+  getSecuritySolutionAlerts,
+} from '../../../../common/lib/alerts';
+import {
+  createSignalsIndex,
+  deleteSignalsIndex,
+  deleteAllRules,
+} from '../../../../../detection_engine_api_integration/utils';
+import { FtrProviderContext } from '../../../../common/ftr_provider_context';
+import { User } from '../../../../common/lib/authentication/types';
+
+import { getPostCaseRequest, postCaseReq, postCommentUserReq } from '../../../../common/lib/mock';
+import {
+  deleteAllCaseItems,
+  deleteCasesByESQuery,
+  deleteCasesUserActions,
+  deleteComments,
+  createCase,
+  createComment,
+  deleteAllComments,
+  superUserSpace1Auth,
+} from '../../../../common/lib/api';
+import {
+  globalRead,
+  noKibanaPrivileges,
+  obsOnly,
+  obsOnlyRead,
+  obsOnlyReadAlerts,
+  obsSec,
+  obsSecRead,
+  secOnly,
+  secOnlyRead,
+  secOnlyReadAlerts,
+  secSolutionOnlyReadNoIndexAlerts,
+  superUser,
+} from '../../../../common/lib/authentication/users';
+
+// eslint-disable-next-line import/no-default-export
+export default ({ getService }: FtrProviderContext): void => {
+  const supertest = getService('supertest');
+  const es = getService('es');
+  const esArchiver = getService('esArchiver');
+  const log = getService('log');
+  const supertestWithoutAuth = getService('supertestWithoutAuth');
+
+  describe('delete_comments', () => {
+    afterEach(async () => {
+      await deleteCasesByESQuery(es);
+      await deleteComments(es);
+      await deleteCasesUserActions(es);
+    });
+
+    describe('happy path', () => {
+      it('should delete all comments', async () => {
+        const postedCase = await createCase(supertest, postCaseReq);
+
+        await createComment({
+          supertest,
+          caseId: postedCase.id,
+          params: postCommentUserReq,
+        });
+
+        const comment = await deleteAllComments({
+          supertest,
+          caseId: postedCase.id,
+        });
+
+        expect(comment).to.eql({});
+      });
+    });
+
+    describe('unhappy path', () => {
+      it('404s when comment belongs to different case', async () => {
+        const postedCase = await createCase(supertest, postCaseReq);
+        await createComment({
+          supertest,
+          caseId: postedCase.id,
+          params: postCommentUserReq,
+        });
+
+        const error = (await deleteAllComments({
+          supertest,
+          caseId: 'fake-id',
+          expectedHttpCode: 404,
+        })) as Error;
+
+        expect(error.message).to.be('No comments found for fake-id.');
+      });
+    });
+
+    describe('alerts', () => {
```

Comment:
If it's not too painful could we have a test that deletes all comments for a case that has some alerts and other comments?

And could we also add an expect that ensures all comments are removed (like delete comments, then do a find and make sure there are no comments for the case or something).